detailed_docs/RC/OnRCStatus.md

```diff
@@ -0,0 +1,147 @@
+## Use Case 1: Notification about RC modules allocation during application’s registration
+
+**Main Flow:**
+
+_Pre-conditions:_
+
+a.  Mobile device is connected to SDL
+
+b.	RC functionality is allowed on HMI
+
+c.	All modules are free
+
+d.	RC functionality is allowed by policies
+
+_Steps:_
+
+1.	RC application starts registration
+
+_Expected:_
+1. SDL sends OnRCStatus with all modules in `freeModules` and empty array of `allocatedModules` to HMI
+2. SDL sends OnRCStatus(allowed = true, freeModules = {}) notification and to registered mobile application
+
+**Alternative flow 1**
+b.1 RC functionality is disallowed on HMI
+b.2 SDL sends OnRCStatus(allowed = false, freeModules = {}, allocatedModules = {}) notification to registered mobile application
+
+**Alternative flow 2**
+1.a.1 Non-RC application starts registration
+1.a.2 SDL does not sends OnRCStatus to HMI and non-RC application
+
+
+## Use Case 2: Notification about RC module allocation after application’s registration
+
+**Main Flow:**
+
+_Pre-conditions:_
+
+a.	Mobile device is connected to SDL
+
+b.	RC functionality is allowed on HMI
+
+c.	All modules are free
+
+d.	RC functionality is allowed by policies
+
+e.	RC app_1 is registered
+
+_Steps:_
+1.	RC app_1 sends control request to module_1 (upon a OnButtonPress or SetInteriorVehicleData request)
+
+2. RC app_1 allocates module_1
+
+_Expected:_
+3. SDL sends OnRCStatus notification to HMI with module_1 in `allocatedModules` parameter and remained modules in `freeModules`
+4. SDL sends OnRCStatus notification to RC app_1 with (allowed = true), module_1 in `allocatedModules` parameter and remained modules in `freeModules`
+
+**Alternative flow 1**
+1.a.1 RC app_1 already allocates module_1
+1.a.2 RC app_1 sends control request to module_1 one more time
+1.a.3 SDL does not send OnRCStatus
+
+**Alternative flow 2**
```

Comment:
I am a little unclear on what this flow is describing.

Reply:
@JackLivio when we try to acquire some module (upon a OnButtonPress or SetInteriorVehicleData request) and HMI doesn't response or response with invalid values of parameters -> module doen't not acquired -> SDL does not sends OnRCStatus